pgrx/src/datum/into.rs

```diff
@@ -352,9 +352,9 @@ impl<'a> IntoDatum for &'a [u8] {
 
             // This is the same as Postgres' `#define SET_VARSIZE_4B` (which have over in
             // `pgrx/src/varlena.rs`), however we're asserting that the input string isn't too big
-            // for a Postgres varlena, since it's limited to 32bits -- in reality it's about half
+            // for a Postgres varlena, since it's limited to 32bits -- in reality it's a quarter
             // that length, but this is good enough
-            debug_assert!(len < (u32::MAX as usize >> 2));
+            assert!(len < (u32::MAX as usize >> 2));
```

Comment:
if we are going to always assert then we should raise it up to before `palloc`, so we don't waste time trying to allocate something we're going to reject anyways.

Reply:
Had that thought, but I assumed that the current place for assertion was intentional. Will fix.

Reply:
nah, I think it just kinda was organic and not thought about heavily because it was a `debug_assert`.